src/component.js

```diff
@@ -169,338 +126,507 @@ class Component {
-        const templateFileName = `${__dirname}/.tln.conf.template`;
-        if (orphan) {
-          const reg = /\/\*[\s\S]*?\*\/|([^:]|^)\/\/.*$/gm;
-          fs.writeFileSync(fileName, fs.readFileSync(templateFileName).toString().replace(reg, ''));
-          this.logger.con('done');
+  /*
+  * Find corresponding component for every ID from array
+  * params:
+  */
+  resolve(ids, resolveEmptyToThis = false) {
+    let result = [];
+    if (ids.length) {
+      ids.forEach((id) => {
+        let component = this;
+        // split id into elements, identify is it absulute path, relative or just standalone id
+        let parts = id.split('/');
+        if (id === '/') {
+          parts = ['', '/']
+        }
+        let strictSearch = true;
+        if (parts.length > 1) {
+          if (parts[0]) {
+            // relative path
+          } else {
+            // absolute path
+            component = this.getRoot();
+            parts.shift();
+          }
         } else {
-          fs.copyFile(templateFileName, fileName, (err) => {
-            if (err) {
-              this.logger.error(err);
-            } else {
-              this.logger.con('done');
-            }
-          });
+          // standalone id
+          strictSearch = false;
+        }
+        // try to find inside child components
+        let e = component.find(parts, strictSearch);
+        if (!(e || strictSearch)) {
+          // try to use components in parent's child
+          e = component.find(parts, strictSearch, component);
         }
+        //
+        if (e) {
+          result.push(e);
+        } else {
+          this.logger.warn('component with id: [', id, '] was not found');
+        }
+      });
+    } else {
+      // resolve to the current folder component
+      if (resolveEmptyToThis) {
+        result.push(this);
       }
     }
+    return result;
   }
 
-  // find entity inside children or pop up to check parent and continue search there
-  find(id, recursive, floatUp = null){
-    this.logger.trace(utils.prefix(this, this.find.name), 'searching', utils.quote(id), 'inside', utils.quote(this.getId()), (floatUp)?'using parent':'within children');
-    let entity = null;
-    if (this.getId() === id) {
-      entity = this;
+
+  /*
+  * Print all information about component
+  * params:
+  */
+  inspectComponent(filter, cntx, yaml, cout) {
+    let r = {};
+    r.id = this.id;
+    r.uuid = this.uuid;
+    r.home = this.home;
+    //r.uuid = this.uuid;
+    r.parent = (this.parent)?(this.parent.id):(null);
+    r.descriptions = [];
+    this.descriptions.forEach( description => {
+      r.descriptions.push({source: description.source, destination: description.destination});
+    });
+    r.tags = [];
+    r.inherits = [];
+    r.depends = [];
+    //
+    const steps = this.findStep('*', filter, this.home, cntx.clone(), []);
+    //
+    r.env = {};
+    const {vars, env} = this.buildEnvironment(this.getVariables());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "env" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://159.65.230.204:9000/coding_rules#rule_key=javascript%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of the unused 'env' variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://159.65.230.204:9000/coding_rules#rule_key=javascript%3AUnusedVariable)